packages/babel-traverse/src/scope/lib/renamer.ts

```diff
@@ -52,63 +49,40 @@ export default class Renamer {
   declare newName: string;
   declare binding: Binding;
 
-  maybeConvertFromExportDeclaration(parentDeclar) {
+  maybeConvertFromExportDeclaration(parentDeclar: NodePath) {
     const maybeExportDeclar = parentDeclar.parentPath;
 
     if (!maybeExportDeclar.isExportDeclaration()) {
       return;
     }
 
-    if (
-      maybeExportDeclar.isExportDefaultDeclaration() &&
-      !maybeExportDeclar.get("declaration").node.id
-    ) {
+    if (maybeExportDeclar.isExportDefaultDeclaration()) {
+      const { declaration } = maybeExportDeclar.node;
+      if (t.isDeclaration(declaration) && !declaration.id) {
+        return;
+      }
+    }
+
+    if (maybeExportDeclar.isExportAllDeclaration()) {
       return;
     }
 
-    splitExportDeclaration(maybeExportDeclar);
+    splitExportDeclaration(
+      maybeExportDeclar as NodePath<
+        Exclude<t.ExportDeclaration, t.ExportAllDeclaration>
+      >,
+    );
   }
 
-  maybeConvertFromClassFunctionDeclaration(path) {
+  maybeConvertFromClassFunctionDeclaration() {
     return; // TODO
-
-    // retain the `name` of a class/function declaration
-
-    // eslint-disable-next-line no-unreachable
-    if (!path.isFunctionDeclaration() && !path.isClassDeclaration()) return;
-    if (this.binding.kind !== "hoisted") return;
-
-    path.node.id = identifier(this.oldName);
-    path.node._blockHoist = 3;
-
-    path.replaceWith(
-      variableDeclaration("let", [
-        variableDeclarator(identifier(this.newName), toExpression(path.node)),
-      ]),
-    );
```

Comment:
Is it possible that we'll need this code in the future? If yes, can we keep it commented out?